tests/lib/util/traverser.js

```diff
@@ -40,4 +40,51 @@ describe("Traverser", () => {
         assert.deepStrictEqual(enteredNodes, [fakeAst, fakeAst.body[0], fakeAst.body[1], fakeAst.body[1].foo]);
         assert.deepStrictEqual(exitedNodes, [fakeAst.body[0], fakeAst.body[1].foo, fakeAst.body[1], fakeAst]);
     });
+
+    it("traverses AST as using 'visitorKeys' option if given", () => {
+        const traverser = new Traverser();
+        const fakeAst = {
+            type: "Program",
+            body: [
+                {
+                    type: "ClassDeclaration",
+                    id: {
+                        type: "Identifier"
+                    },
+                    superClass: null,
+                    body: {
+                        type: "ClassBody",
+                        body: []
+                    },
+                    experimentalDecorators: [
+                        {
+                            type: "Decorator",
+                            expression: {}
+                        }
+                    ]
+                }
+            ]
+        };
+
+        fakeAst.body[0].parent = fakeAst;
+
+        const visited = [];
+
+        // default.
+        traverser.traverse(fakeAst, {
+            enter: node => visited.push(node.type)
+        });
+        assert.deepStrictEqual(visited, ["Program", "ClassDeclaration", "Identifier", "ClassBody"]);
+
+        visited.splice(0, visited.length);
+
+        // with keys option.
```

Comment:
Could this test be split in two? (Or, do we even need the first half of this test?)